### PR TITLE
Remove assert side effect

### PR DESCRIPTION
**Pull request by tarruda**
Put the relevant code/debug var undef #ifndef NDEBUG.

### CODE REVIEW

File: lmpack.c [hunk 1]
```diff
@@ -237,7 +240,10 @@ static mpack_uint32_t lmpack_objlen(lua_State *L, int *is_array)
   if ((size_t)-1 > (mpack_uint32_t)-1 && len > (mpack_uint32_t)-1)
     /* msgpack spec doesn't allow lengths > 32 bits */
     len = (mpack_uint32_t)-1;
-  assert(top == lua_gettop(L));
```
Comment (justinmk) [inline lmpack.c]: This one uses `==` so it doesn't need to be changed.

**Comment by justinmk** [comment]
@tarruda this could be merged

**Comment by justinmk** [comment]
@tarruda friendly ping

**Comment by tarruda** [comment]
Merged and released a new version. Sorry for the long delay.

**Pull request merged**